subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/derived/JvmDerivedVariantsIntegrationTest.groovy

```diff
@@ -171,6 +180,78 @@ task resolve {
         noExceptionThrown()
     }
 
+    def "can re-select artifacts if dependency has explicit artifact when lenient=true"() {
+        file("consumer/build.gradle") << """
+dependencies {
+    implementation 'com.example:test'
+    implementation 'com.example:test:1.0:foo'
+}
+
+task resolve {
+    def normal = configurations.runtimeClasspath.incoming.files
+    def reselected = configurations.runtimeClasspath.incoming.artifactView {
+        withVariantReselection()
+        lenient = true
```

Comment:
🤔 I think with the changes you've made, it no longer matters if lenient is true or false, you won't get an exception